forge/src/runner.rs

```diff
@@ -369,6 +437,147 @@ impl<'a> ContractRunner<'a> {
         })
     }
 
+    #[tracing::instrument(name = "invariant-test", skip_all)]
+    pub fn run_invariant_test(
+        &mut self,
+        runner: TestRunner,
+        setup: TestSetup,
+        test_options: TestOptions,
+        functions: Vec<&Function>,
+        known_contracts: Option<&BTreeMap<ArtifactId, (Abi, Vec<u8>)>>,
+        identified_contracts: BTreeMap<Address, (String, Abi)>,
+    ) -> Result<Vec<TestResult>> {
+        let empty = BTreeMap::new();
+        let project_contracts = known_contracts.unwrap_or(&empty);
+        let TestSetup { address, logs, traces, labeled_addresses, .. } = setup;
+
+        let start = Instant::now();
+        let prev_db = self.executor.backend().db.clone();
+        let mut evm = InvariantExecutor::new(
+            &mut self.executor,
+            runner,
+            &identified_contracts,
+            project_contracts,
+        );
+
+        let invariant_contract =
+            InvariantContract { address, invariant_functions: functions, abi: self.contract };
+
+        if let Some(InvariantFuzzTestResult { invariants, cases, reverts }) = evm.invariant_fuzz(
+            invariant_contract,
+            InvariantTestOptions {
+                depth: test_options.invariant_depth,
+                fail_on_revert: test_options.invariant_fail_on_revert,
+                call_override: test_options.invariant_call_override,
+            },
+        )? {
+            let results = invariants
+                .iter()
+                .map(|(_k, test_error)| {
+                    let mut counterexample_sequence = vec![];
+                    let mut logs = logs.clone();
+                    let mut traces = traces.clone();
+
+                    if let Some(ref error) = test_error {
+                        // we want traces for a failed fuzz
+                        let mut ided_contracts = identified_contracts.clone();
+                        if let TestError::Fail(_reason, vec_addr_bytes) = &error.test_error {
+                            // Reset DB state
+                            self.executor.backend_mut().db = prev_db.clone();
+                            self.executor.set_tracing(true);
```

Comment:
this is no longer possible on master

not really understand this, so hard to comment how to fix